### PR TITLE
documentation-updates

### PR DESCRIPTION
**Pull request by pgladkows**

### CODE REVIEW

File: docs/developer_guide.md [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-- docker
+- Docker
 - opm https://github.com/operator-framework/operator-registry/releases
 - operator-sdk https://github.com/operator-framework/operator-sdk/releases
+- olm https://github.com/operator-framework/operator-lifecycle-manager/releases
 - golang 1.19
 - make
-- kubectl
-- ubuntu20 or redhat8 OS
+- Ubuntu20 or RedHat8 OS
```
Comment (dtrawins) [inline docs/developer_guide.md:10]:
```suggestion
- Ubuntu22 or RedHat8 OS
```

File: tests/README.md [hunk 2]
```diff
@@ -1,26 +1,84 @@
...
+sudo apt-get update
+```
+
+Then, install containerd:
+
+```
+sudo apt remove containerd
+sudo apt update
+sudo apt install containerd.io
+sudo rm /etc/containerd/config.toml
```
Comment (dtrawins) [inline tests/README.md:44]: why is it removed?

File: docs/developer_guide.md [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 
 ## Prerequisites
-- docker
+- Docker
 - opm https://github.com/operator-framework/operator-registry/releases
 - operator-sdk https://github.com/operator-framework/operator-sdk/releases
+- olm https://github.com/operator-framework/operator-lifecycle-manager/releases
 - golang 1.19
 - make
-- kubectl
```
Comment (mzegla) [inline docs/developer_guide.md:9]: So it's not needed in any of mentioned make targets?

File: tests/README.md [hunk 4]
```diff
@@ -1,26 +1,84 @@
 ## Preparing K8S environment for tests
```
Comment (mzegla) [inline tests/README.md:1]: Maybe we should mention that it's an instruction for Ubuntu 22?

File: tests/README.md [hunk 5]
```diff
@@ -1,26 +1,84 @@
...
+```
+set -e
+sudo apt-get update
+sudo apt-get install -y apt-transport-https ca-certificates curl gpg
+sudo mkdir -p -m 755 /etc/apt/keyrings
+curl -fsSL https://pkgs.k8s.io/core:/stable:/v1.28/deb/Release.key | sudo gpg --dearmor -o /etc/apt/keyrings/kubernetes-apt-keyring.gpg
+echo 'deb [signed-by=/etc/apt/keyrings/kubernetes-apt-keyring.gpg] https://pkgs.k8s.io/core:/stable:/v1.28/deb/ /' | sudo tee /etc/apt/sources.list.d/kubernetes.list
+sudo apt-get update
+sudo apt-get install -y kubelet=1.28.7-1.1 kubeadm=1.28.7-1.1 kubectl=1.28.7-1.1
+sudo apt-mark hold kubelet kubeadm kubectl
```
Comment (mzegla) [inline tests/README.md:18]: Maybe adding `-y` to update steps would also be useful?

File: tests/README.md [hunk 6]
```diff
@@ -1,26 +1,84 @@
...
+echo "CLI tools installed"
+```
+
+2. Setup containerd
+
+Set up Docker's apt repository as described in [Docker documentation](https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository).
+```
+sudo apt-get update
```
Comment (mzegla) [inline tests/README.md:26]: If we assume this point shall always be run with the 1., then this update does not seem necessary

File: tests/README.md [hunk 7]
```diff
@@ -1,26 +1,84 @@
...
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+  $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
+  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+sudo apt-get update
+```
+
+Then, install containerd:
+
+```
+sudo apt remove containerd
```
Comment (mzegla) [inline tests/README.md:41]: Perhaps we could move the cleanup phase to the beginning like: 
"Clean up old versions of containerd" and then proceed with the fresh setup and installation?

File: tests/README.md [hunk 8]
```diff
@@ -1,26 +1,84 @@
...
+sudo apt update
+sudo apt install containerd.io
+sudo rm /etc/containerd/config.toml
+sudo systemctl restart containerd
+```
+
+2. Initialize a Kubernetes control-plane node
```
Comment (mzegla) [inline tests/README.md:48]:
```suggestion
3. Initialize a Kubernetes control-plane node
```

File: tests/README.md [hunk 9]
```diff
@@ -1,26 +1,84 @@
...
 sudo chown ${id} ${HOME}/.kube/config
 kubectl apply -f https://github.com/coreos/flannel/raw/master/Documentation/kube-flannel.yml
-kubectl taint nodes --all node-role.kubernetes.io/master-
+kubectl taint nodes --all node-role.kubernetes.io/control-plane:NoSchedule-
 kubectl get pod --all-namespaces
 echo "cluster installed"
 ```
 
-Install OLM
+3. Install OLM in K8S cluster manually
```
Comment (mzegla) [inline tests/README.md:63]:
```suggestion
4. Install OLM in K8S cluster manually
```

File: tests/README.md [hunk 10]
```diff
@@ -1,26 +1,84 @@
...
 ```
-echo "installing olm"
-operator-sdk olm install
+operator-sdk olm install --version v0.27.0
 operator-sdk olm status
+echo "olm installed"
```
Comment (mzegla) [inline tests/README.md:67]:
```suggestion
echo "OLM installed"
```

File: tests/README.md [hunk 11]
```diff
@@ -1,26 +1,84 @@
...
 ```
 
-Configure ImageStream and BuildConfig CRDs
+or using Makefile
+
 ```
-kubectl apply -f tests/os-crds.yaml
+cd ..
+make cluster_clean
+echo "olm installed"
```
Comment (mzegla) [inline tests/README.md:75]:
```suggestion
echo "OLM installed"
```

File: tests/README.md (hunk 4 above)
Comment (pgladkows) [reply tests/README.md:1]: For now, this has only been tested on Ubuntu20

File: docs/developer_guide.md (hunk 1 above)
Comment (pgladkows) [reply docs/developer_guide.md:10]: For now, this has only been tested on Ubuntu20

File: docs/developer_guide.md (hunk 3 above)
Comment (pgladkows) [reply docs/developer_guide.md:9]: My mistake, it is, I'll bring it back.

File: tests/README.md (hunk 2 above)
Comment (pgladkows) [reply tests/README.md]: This is required if any version of containerd was previously installed. I moved it to the cleanup section.

File: tests/README.md (hunk 5 above)
Comment (pgladkows) [reply tests/README.md]: added

File: tests/README.md (hunk 6 above)
Comment (pgladkows) [reply tests/README.md]: removed

File: tests/README.md (hunk 7 above)
Comment (pgladkows) [reply tests/README.md]: added

File: tests/README.md (hunk 8 above)
Comment (pgladkows) [reply tests/README.md]: fixed

File: tests/README.md (hunk 9 above)
Comment (pgladkows) [reply tests/README.md]: fixed

File: tests/README.md (hunk 10 above)
Comment (pgladkows) [reply tests/README.md]: fixed

File: tests/README.md (hunk 11 above)
Comment (pgladkows) [reply tests/README.md]: fixed

**Review by ngrozae — APPROVED** [review]

File: docs/developer_guide.md (hunk 1 above)
Comment (pgladkows) [reply docs/developer_guide.md]: update: it also works on Ubuntu 22

File: tests/README.md (hunk 4 above)
Comment (pgladkows) [reply tests/README.md:1]: Update: it also works on Ubuntu 22. I've added NOTE at the beginning.

**Review by mzegla — APPROVED** [review]

**Review by dtrawins — APPROVED** [review]

**Pull request merged**